vm/src/obj/objstr.rs

```diff
@@ -519,6 +522,30 @@ impl PyString {
         }
     }
 
+    /// Return true if all characters in the string are printable or the string is empty,
+    /// false otherwise.  Nonprintable characters are those characters defined in the
+    /// Unicode character database as `Other` or `Separator`,
+    /// excepting the ASCII space (0x20) which is considered printable.
+    ///
+    /// All characters except those characters defined in the Unicode character
+    /// database as following categories are considered printable.
+    ///   * Cc (Other, Control)
+    ///   * Cf (Other, Format)
+    ///   * Cs (Other, Surrogate)
+    ///   * Co (Other, Private Use)
+    ///   * Cn (Other, Not Assigned)
+    ///   * Zl Separator, Line ('\u2028', LINE SEPARATOR)
+    ///   * Zp Separator, Paragraph ('\u2029', PARAGRAPH SEPARATOR)
+    ///   * Zs (Separator, Space) other than ASCII space('\x20').
+    #[pymethod]
+    fn isprintable(&self, _vm: &VirtualMachine) -> bool {
+        self.value.is_empty()
+            || self.value.chars().all(|c| match c {
```

Comment:
Should this be a `||`?

Reply:
`"".isprintable()` return `True`, I will remove empty check because `Iterator::all` returns true if the iterator has no elements.